### PR TITLE
App Submission: Immich

### PR DESCRIPTION
**Pull request by levma**
# App Submission

### App name
Immich

### 256x256 SVG icon
[Logo](https://immich.app/immich-logo.svg)

### Gallery images
- [Home Dashboard](https://github.com/immich-app/immich/raw/main/design/web-home.jpeg)
- [Image View](https://github.com/immich-app/immich/raw/main/design/web-detail.jpeg)
- [Shared Albums](https://imgur.com/a/GxCMgN0)

### I have tested my app on:
- [ ] [Umbrel dev environment](https://github.com/getumbrel/umbrel-dev)
- [x] [Umbrel OS on a Raspberry Pi 4](https://github.com/getumbrel/umbrel-os)
- [ ] [Custom Umbrel install on Linux](https://github.com/getumbrel/umbrel#-installation)

### CODE REVIEW

File: immich/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,129 @@
+version: "3.7"
+
+services:
+  app_proxy:
+    environment:
+      APP_HOST: immich-proxy
+      APP_PORT: 8080
+      PROXY_AUTH_ADD: "false"
```
Comment (nevets963) [inline immich/docker-compose.yml:8]: Is this turned off to allow third party apps to talk to immich's APi?

**Review by nevets963 — COMMENTED** [review]
@levma Many many thanks for submitting Immich into the Umbrel App Store!

I just gave it a test and it's a really nicely built app with a decent UI.

I've left some feedback on this PR and I think we get this approved pretty quickly.

File: immich/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+      REDIS_HOSTNAME: immich-redis
+      LOG_LEVEL: ${APP_IMMICH_LOG_LEVEL}
+      JWT_SECRET: ${APP_SEED}
+      DISABLE_REVERSE_GEOCODING: ${APP_IMMICH_DISABLE_REVERSE_GEOCODING}
+      REVERSE_GEOCODING_PRECISION: ${APP_IMMICH_REVERSE_GEOCODING_PRECISION}
+      PUBLIC_LOGIN_PAGE_MESSAGE: ${APP_IMMICH_PUBLIC_LOGIN_PAGE_MESSAGE}
+    depends_on:
+      - redis
+      - postgres
+    restart: always
```
[the remark this answers is not in the document]
Comment (levma) [reply immich/docker-compose.yml:32]: no, this was just copy-pasted -> i'll change it to `on-failure` …

File: immich/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+services:
+  app_proxy:
+    environment:
+      APP_HOST: immich-proxy
+      APP_PORT: 8080
+      PROXY_AUTH_ADD: "false"
+
+  immich-server:
+    container_name: immich-server
+    hostname: immich-server
```
[the remark this answers is not in the document]
Comment (levma) [reply immich/docker-compose.yml:12]: No, this was just "easier" from my point of view. I'll remove the lines and will change the env vars accordingly to `app_service_1`

File: immich/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    image: "altran1502/immich-server:${APP_IMMICH_VERSION}"
```
[the remark this answers is not in the document]
Comment (levma) [reply immich/docker-compose.yml:13]: Yes, i'll add them

File: immich/docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+      POSTGRES_DB: ${APP_IMMICH_DB_DATABASE_NAME}
+      PG_DATA: /var/lib/postgresql/data
+    volumes:
+      - ${APP_DATA_DIR}/pgdata:/var/lib/postgresql/data
+    restart: always
+
+  immich-proxy:
+    container_name: immich-proxy
+    hostname: immich-proxy
+    image: altran1502/immich-proxy:release
```
[the remark this answers is not in the document]
Comment (levma) [reply immich/docker-compose.yml:122]: Oh, I missed that one. Sure

File: immich/docker-compose.yml (hunk 1 above)
Comment (levma) [reply immich/docker-compose.yml:8]: Yes, Immich does have mobile apps for iOS and Android. Immich does also require a login. So I came to the conclusion that an additional auth middleware wasnt needed. Is this correct?

Thank you for your review and feedback!!

File: immich/docker-compose.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (levma) [reply immich/docker-compose.yml:12]: Maybe it is needed at least for immich-server and immich-web as the immich-proxy refers to hard coded domain names, see https://github.com/immich-app/immich/blob/948ff5530ca063c877e1e4b3b5f6f2c51111bc19/nginx/nginx.conf#L90
I will try with and without hostname config entries

[the remark this answers is not in the document]
Comment (levma) [reply immich/docker-compose.yml:12]: Ok, I had to re-add the hostnames for the api and web-ui as the internal proxy didnt work without them

[the remark this answers is not in the document]
Comment (nevets963) [reply immich/docker-compose.yml]: It looks like immich has been updated to allow env. vars. to be set to set the web and server urls: https://github.com/immich-app/immich/blob/fd15cdbf40b1d98c01fc90b840078318156e37c7/docker/docker-compose.yml#L81-L84 I've updated this PR to reflect that and now giving it a test.

[the remark this answers is not in the document]
Comment (nevets963) [reply immich/docker-compose.yml]: That seems to have worked.

**Review by nevets963 — APPROVED** [review]
@levma Many thanks again for submitting Immich into the Umbrel App Store!

I went through the remaining niggles today and looks like it's now working really nicely.

Tested on `arm64` and `amd64`.

Tested using iOS app and managed to backup some photos.

This is approved from me 🚀 

I'll need @mayankchhabra's help for the app store graphics (where they'll live here: https://github.com/getumbrel/umbrel-apps-gallery). Once we have those, then this is good to merge in.

**Comment by abjea** [comment]
@levma @nevets963 

any plans to get this updated to the latest versions?  i am trying to do a bulk upload using CLI but it needs an API key---newer versions allow you to obtain API key in the user settings

thanks

**Comment by rpjcbs** [comment]
I second what @abjea has said.

@levma & @nevets963 any scope to getting Immich updated? Looks really good and can’t wait to give it a try.

**Comment by nmfretz** [comment]
@rpjcbs an update is on its way! See open PR here: https://github.com/getumbrel/umbrel-apps/pull/526

**Comment by BubbyWoodz** [comment]
Please Update the app to v1.132.0

**Comment by al-lac** [comment]
@BubbyWoodz will be available once https://github.com/getumbrel/umbrel-apps/pull/2651 is merged.

No need to comment, we usually get to it in a few days.

**Pull request merged**